### PR TITLE
fix: contract bundles have duplicate dependencies

### PR DESCRIPTION
**Pull request by 0xpatrickdev**
- [x] forces npm to resolve one version for each agoric-sdk dependency 3c1cc85
- [x] adds `yarn build` to ci

| Scenario  | Bundle ID                     | File Size |
|-----------|-------------------------------|-----------|
| **Before** | b1-1e1aeca....ad8e572 | 1.8M |
| | b1-34a17f3....40810d | 1.6M |
| **After** | b1-60228f8....21ae8 | 747K |
| | b1-2767ddb....07b8b | 584K |

I also tested with the changes in #28:

```diff
diff --git a/contract/src/gameAssetContract.js b/contract/src/gameAssetContract.js
index 90c0cfc..63680e9 100644
--- a/contract/src/gameAssetContract.js
+++ b/contract/src/gameAssetContract.js
@@ -5,6 +5,7 @@ import { Far } from '@endo/far';
 import { M, getCopyBagEntries } from '@endo/patterns';
 import { AmountMath, AssetKind } from '@agoric/ertp/src/amountMath.js';
 import { AmountShape } from '@agoric/ertp/src/typeGuards.js';
+import { assertIssuerKeywords } from "@agoric/zoe/src/contractSupport/index.js";
 // Use the deprecated atomicRearrange API
 // for compatibility with mainnet1B.
 import '@agoric/zoe/exported.js';
@@ -27,6 +28,7 @@ const bagValueSize = amt => {
  * @param {ZCF<{joinPrice: Amount}>} zcf
  */
 export const start = async zcf => {
+  assertIssuerKeywords(zcf, harden(['Price']));
   const { joinPrice } = zcf.getTerms();

   const { zcfSeat: gameSeat } = zcf.makeEmptySeatKit();
``` 

And am observing smaller bundles:

| Scenario  | Bundle ID                     | File Size |
|-----------|-------------------------------|-----------|
| **Before** | b1-e14ac9e....f0052 | 4.9M |
| | b1-96191af....04cd8a | 1.6M |
| **After** | b1-90efd17....3ca9 | 1.5M |
| | b1-2767ddb....07b8b | 584K |

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -18,5 +18,7 @@ jobs:
         run: yarn
       - name: yarn lint
         run: yarn lint
+      - name: yarn build
+        run: yarn build
```
Comment (dckc) [inline .github/workflows/pr.yml:22]: We should test that the bundle, when compressed, is <1MB.
Want to do it in this PR or leave it for later?

File: contract/package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     "docker:make": "docker compose exec agd make -C /workspace/contract",
     "make:help": "make list",
     "start": "yarn docker:make clean start-contract print-key",
-    "build": "exit 0",
+    "build": "agoric run scripts/build-game1-start.js",
```
Comment (dckc) [inline contract/package.json:14]: I'm a little nervous about permissions conflicting with doing this build in docker, but I guess it's OK for now.

File: package.json [hunk 3]
```diff
@@ -9,6 +9,28 @@
     "ui"
   ],
   "resolutions": {
```
Comment (dckc) [inline package.json:11]:
```suggestion
  "resolutions-note": "work-around for https://github.com/Agoric/agoric-sdk/issues/8621",
  "resolutions": {
```

also, to get a back-link from that issue:
 - https://github.com/Agoric/agoric-sdk/issues/8621

File: yarn.lock [hunk 4]
```diff
@@ -1381,31 +1338,6 @@
     "@endo/zip" "^0.2.31"
     ses "^0.18.4"
 
-"@endo/compartment-mapper@^0.8.4", "@endo/compartment-mapper@^0.8.5":
```
Comment (dckc) [inline yarn.lock:1384]: :clap: 

many fewer dependencies

**Review by dckc — APPROVED** [review]
I suggest documenting that the resolutions are a work-around.

File: .github/workflows/pr.yml (hunk 1 above)
Comment (0xpatrickdev) [reply .github/workflows/pr.yml:22]: Our heads are in the [same place](https://github.com/Agoric/agoric-sdk/pull/8559#issuecomment-1876111342)!

I think a separate PR is best. I also wonder if `agoric run` / `writeCoreProposal` should be responsible for enforcing this (or, at least warning)

Comment (0xpatrickdev) [reply .github/workflows/pr.yml:22]: New issue here: https://github.com/Agoric/dapp-offer-up/issues/36. Can move it to `agoric-sdk` if you agree this should be a responsibility of `agoric run`

**Pull request merged**